fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java

```diff
@@ -624,6 +628,16 @@ protected DruidFilterBuilder buildDruidFilterBuilder() {
     }
 
     /**
+     * Creates an object that generates map of Api Having from having string.
+     * Constructs a {@link DefaultHavingApiGenerator} by default.
+     *
+     * @return An object to generate having maps from having string.
+     */
+    protected HavingGenerator buildHavingGenerator() {
+        return new DefaultHavingApiGenerator(loader.getMetricDictionary());
```

Comment:
Typically, we pass in the dictionaries as explicit method parameters, rather than calling `loader::get`.

Reply:
As discussed, passing the loader to the buildHavingGenerator instead of passing metric dictionary to the DefaultHavingApiGenerator constructor to have flexibility in the future.